File: extract.js
const fs = require('fs');

function parseSnapshot(filePath, outputPath, uidPrefix) {
    const text = fs.readFileSync(filePath, 'utf-8');
    const lines = text.split('\n');
    const codeFragments = [];
    const pattern = new RegExp(`uid=${uidPrefix}_\\d+\\s+StaticText\\s+"(.*)"$`);
    
    for (const line of lines) {
        const match = line.match(pattern);
        if (match) {
            let content = match[1].replace(/\\"/g, '"').replace(/\\n/g, '\n');
            if (content.trim().startsWith('//')) {
                content = `/* ${content.trim().substring(2)} */ `;
            }
            codeFragments.push(content);
        }
    }

    fs.writeFileSync(outputPath, codeFragments.join(''), 'utf-8');
}

parseSnapshot('C:/Users/hreya/.gemini/antigravity/brain/7c7c9d18-007b-41b5-be56-325b04d48c6d/.tempmediaStorage/snapshot_full_1786264768871.txt', 'Component.tsx', '11');
parseSnapshot('C:/Users/hreya/.gemini/antigravity/brain/7c7c9d18-007b-41b5-be56-325b04d48c6d/.tempmediaStorage/snapshot_full_1786264827507.txt', 'Usage.tsx', '20');

File: extract.py
import re

def parse_snapshot(file_path, output_path, uid_prefix):
    with open(file_path, 'r', encoding='utf-8') as f:
        text = f.read()

    lines = text.split('\n')
    code_fragments = []
    # match uid=<prefix>_<num> StaticText "<content>"
    pattern = re.compile(rf'uid={uid_prefix}_\d+\s+StaticText\s+"(.*)"$')
    
    for line in lines:
        m = pattern.search(line)
        if m:
            content = m.group(1).replace('\\"', '"').replace('\\n', '\n')
            code_fragments.append(content)

    with open(output_path, 'w', encoding='utf-8') as f:
        f.write(''.join(code_fragments))

# Parse Component.tsx
parse_snapshot('C:/Users/hreya/.gemini/antigravity/brain/7c7c9d18-007b-41b5-be56-325b04d48c6d/.tempmediaStorage/snapshot_full_1786264768871.txt', 'Component.tsx', '11')

# Parse Usage.tsx
parse_snapshot('C:/Users/hreya/.gemini/antigravity/brain/7c7c9d18-007b-41b5-be56-325b04d48c6d/.tempmediaStorage/snapshot_full_1786264827507.txt', 'Usage.tsx', '20')
